itinerary_generator/parser.py

```
@@ -0,0 +1,101 @@
+        
+    Raises:
+        KeyError: If required date fields are missing
+        ValueError: If dates are in an invalid format
+    """
+    try:
+        start = datetime.fromisoformat(trip["startDate"].replace("Z", "+00:00"))
+        end = datetime.fromisoformat(trip["endDate"].replace("Z", "+00:00"))
+        return start, end
+    except KeyError as e:
+        # Specific error for missing required fields
+        raise KeyError(f"Trip data missing required date field: {e}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)